packages/components/pie-notification/src/index.ts

```diff
@@ -1,24 +1,193 @@
+import '@justeattakeaway/pie-icons-webc/IconAlertCircle';
+import '@justeattakeaway/pie-icons-webc/IconAlertTriangle';
+import '@justeattakeaway/pie-icons-webc/IconCheckCircle';
 
 // Valid values available to consumers
 export * from './defs';
 
 const componentSelector = 'pie-notification';
+const componentClass = 'c-notification';
 
 /**
  * @tagname pie-notification
  */
 export class PieNotification extends LitElement implements NotificationProps {
-    render () {
-        return html`<h1 data-test-id="pie-notification">Hello world!</h1>`;
+    @property({ type: Boolean })
+    public isOpen = false;
+
+    @property()
+    @validPropertyValues(componentSelector, variants, 'neutral')
+    public variant: NotificationProps['variant'] = 'neutral';
+
+    @property({ type: Boolean })
+    public compact = false;
+
+    @property({ type: String })
+    public heading!: string;
+
+    @property()
+    @validPropertyValues(componentSelector, headingLevels, 'h2')
+    public headingLevel: NotificationProps['headingLevel'] = 'h2';
+
+    @property({ type: Boolean })
+    public hideIcon = false;
+
+    @property({ type: Boolean })
+    public hideCloseIcon = false;
+
+    @property({ type: Boolean })
+    private _hasExternalIcon = false;
+
+    @query('slot[name="icon"]')
+    private _iconSlot?: HTMLSlotElement;
+
+    protected firstUpdated (): void {
+        if (this._iconSlot) {
+            const assignedNodes = this._iconSlot.assignedNodes();
+
+            if (assignedNodes.length > 0) {
+                this._hasExternalIcon = true;
+            }
+        }
     }
 
     // Renders a `CSSResult` generated from SCSS by Vite
     static styles = unsafeCSS(styles);
+
+    /**
+     * Template for the footer area
+     * Called within the main render function.
+     *
+     * @private
+     */
+    private renderFooter () {
+        return html`
+            <footer class="${componentClass}-footer"></footer>
+        `;
+    }
+
+    /**
+     * Template for the header area
+     * Called within the main render function.
+     *
+     * @param {NotificationProps['heading']} heading
+     * @param {StaticValue} headingTag
+     *
+     * @private
+     */
+    private renderNotificationHeading (heading: NotificationProps['heading'], headingTag: StaticValue): TemplateResult {
+        return html`
+            <header class="${componentClass}-header" data-test-id="${componentSelector}-header">
+                <${headingTag} class="${componentClass}-heading" data-test-id="${componentSelector}-heading">${heading}</${headingTag}>
+            </header>
+        `;
+    }
+
+    /**
+     * Util static function that returns a template with a default icon according to the chosen variant.
+     * Called within the renderIcon method.
+     *
+     * @param {boolean} variant
+     *
+     * @static
+     */
+    static renderIconVariant (variant: NotificationProps['variant']) {
+        let iconVariant = null;
```

Comment:
Would it be more appropriate to return Lit's [`nothing`](https://lit.dev/docs/v1/lit-html/writing-templates/#rendering-nothing) value here?

Reply:
You're right.

Reply:
I took this as opportunity to refactor this method as well